tests/cuda_test/test_print_deallocate.py

```diff
@@ -0,0 +1,40 @@
+# pylint: disable=missing-function-docstring, missing-module-docstring/
+# pylint: disable=protected-access
+import pytest
+from pyccel.ast.variable import Variable
+from pyccel.codegen.printing.ccudacode import CcudaCodePrinter
+from pyccel.ast.core      import Deallocate
+
+@pytest.mark.parametrize( 'language', [
+        pytest.param("ccuda", marks = pytest.mark.ccuda)
+    ]
+)
+def test_print_deallocate(language):
+    # Test case 1: variable is in host memory
+    var = Variable('int', 'y', memory_location='host')
+    expr = Deallocate(var)
+    printer = CcudaCodePrinter('')
+    output = printer._print_Deallocate(expr)
+    expected_output = "cuda_free_host(y);\n"
+    assert output == expected_output
+    # Test case 2: variable is in device memory
+    var = Variable('int', 'y', memory_location='device')
+    expr = Deallocate(var)
+    printer = CcudaCodePrinter('')
+    output = printer._print_Deallocate(expr)
+    expected_output = "cuda_free(y);\n"
+    assert output == expected_output
+    #Test case 3: Memory location of variable is managed
+    var = Variable('int', 'y', memory_location='managed')
+    expr = Deallocate(var)
+    printer = CcudaCodePrinter('')
+    output = printer._print_Deallocate(expr)
+    expected_output = "cuda_free(y);\n"
+    assert output == expected_output
+    #Test case 4: variable is an alias
+    var = Variable('int', 'y', memory_handling='alias')
+    expr = Deallocate(var)
+    printer = CcudaCodePrinter('')
+    output = printer._print_Deallocate(expr)
+    expected_output = "cuda_free_pointer((*y));\n"
+    assert output == expected_output
```

Comment:
I assume this test was added to avoid the Coverage Checker failure? I don't think this is a good idea. It tests the code which is good. But not having an error indicating that this code isn't tested hides the fact that Pyccel doesn't  seem to be generating a `Deallocate` node when the code requires it. I would prefer that the memory leak was fixed rather than just ensuring the printer code is tested

Reply:
That said, this definitely should fix the coverage checker and it doesn't seem to. I'm going to play around with your branch briefly and see if I can find the problem

Reply:
Yes, the tests were added to address the Coverage Checker failure. Regarding the memory leaks tests, you are correct, and I'll work on fixing it.

Reply:
By "fix the memory leak" I meant that if the deallocate node isn't being printed then there is probably a memory leak there. But don't worry about it. It looks like the coverage doesn't work properly. I'm working on fixing it

Reply:
Oh I see. I thought you were talking about adding tests to check the memory leaks ...